pkg/kv/kvserver/rangefeed/buffered_sender.go

```diff
@@ -121,7 +126,8 @@ func (bs *BufferedSender) run(
 			return nil
 		case <-bs.notifyDataC:
 			for {
-				e, success := bs.popFront()
+				e, success, remaining := bs.popFront()
+				bs.metrics.BufferedSenderQueueSize.Update(remaining)
```

Comment:
I don't think you want `Update` here because we might have multiple buffered senders.

Reply:
Ahh, good catch. That explains the unexpected numbers in some of my early runs.